homeassistant/components/owlet/sensor.py

```diff
@@ -0,0 +1,130 @@
+"""
+Support for Owlet sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.owlet/
+"""
+from datetime import timedelta
+
+from homeassistant.components.owlet import DOMAIN as OWLET_DOMAIN
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import dt as dt_util
+
+SCAN_INTERVAL = timedelta(seconds=120)
+
+SENSOR_CONDITIONS = {
+    'oxygen_level': {
+        'name': 'Oxygen Level',
+        'device_class': ''
+    },
+    'heart_rate': {
+        'name': 'Heart Rate',
+        'device_class': ''
+    }
+}
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up owlet binary sensor."""
+    device = hass.data[OWLET_DOMAIN]
+
+    entities = []
+    for condition in SENSOR_CONDITIONS:
+        if condition in device.monitor:
+            entities.append(OwletSensor(device, condition))
+
+    add_entities(entities, True)
+
+
+class OwletSensor(Entity):
+    """Representation of Owlet sensor."""
+
+    def __init__(self, device, condition):
+        """Init owlet binary sensor."""
+        self._device = device
+        self._condition = condition
+        self._state = None
+        self._prop_expiration = None
+        self._is_charging = None
+        self._battery_level = None
+        self._sock_off = None
+        self._sock_connection = None
+        self._movement = None
+
+    @property
+    def name(self):
+        """Return sensor name."""
+        return '{} {}'.format(self._device.name,
+                              SENSOR_CONDITIONS[self._condition]['name'])
+
+    @property
+    def state(self):
+        """Return current state of sensor."""
+        return self._state
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return SENSOR_CONDITIONS[self._condition]['device_class']
+
+    @property
+    def is_charging(self):
```

Comment:
Remove all these properties that are not part of the base entity class and just returns the attribute.